src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/PagedBufferedTextWriter.cs

```diff
@@ -83,11 +83,21 @@ public override void Write(char value)
 
         public override void Write(char[] buffer)
         {
+            if (buffer == null)
+            {
+                return;
+            }
+
             Write(buffer, 0, buffer.Length);
         }
 
         public override void Write(char[] buffer, int index, int count)
         {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
```

Comment:
@rynowak could you to confirm that having this throw is the correct behavior? It seems pretty dangerous because it causes the response to fail after we've already sent the headers but @dougbu is correct that that's what we do in System.IO.TextWriter.

P.S.: Sorry for the dupe, I accidentally staled the other comment to this effect.